Client/Frontend/Shields/ShareTrackersController.swift

```diff
@@ -0,0 +1,339 @@
+            case .trackerAdWarning:
+                return Strings.ShieldEducation.trackerAdWarningSubtitle
+            case .videoAdBlock:
+                return Strings.ShieldEducation.videoAdBlockSubtitle
+            case .trackerAdCountBlock:
+                return Strings.ShieldEducation.trackerAdCountBlockSubtitle
+            case .encryptedConnectionWarning:
+                return Strings.ShieldEducation.encryptedConnectionWarningSubtitle
+        }
+    }
+    
+    var actionTitle: String {
+        switch self {
+            case .trackerCountShare,
+                 .videoAdBlock,
+                 .trackerAdCountBlock,
+                 .encryptedConnectionWarning:
+                return Strings.ShieldEducation.educationDismissTitle
+            case .trackerAdWarning:
+                return Strings.ShieldEducation.educationInspectTitle
+        }
+    }
+}
+
+// MARK: - ShareTrackersController
+
+class ShareTrackersController: UIViewController, Themeable, PopoverContentComponent {
+    
+    // MARK: Action
+    
+    enum Action {
+        case takeALookTapped
+        case dontShowAgainTapped
+        case shareEmailTapped
+        case shareTwitterTapped
+        case shareFacebookTapped
+        case shareMoreTapped
+    }
+    
+    // MARK: Properties
+    
+    private let theme: Theme
+    private let trackingType: TrackingType
+    
+    private let shareTrackersView: ShareTrackersView
+    
+    private lazy var gradientView = GradientView(
+        colors: [#colorLiteral(red: 0.968627451, green: 0.2274509804, blue: 0.1098039216, alpha: 1), #colorLiteral(red: 0.7490196078, green: 0.07843137255, blue: 0.6352941176, alpha: 1)],
+        positions: [0, 1],
+        startPoint: .zero,
+        endPoint: CGPoint(x: 1, y: 0.5))
+    
+    var actionHandler: ((Action) -> Void)?
+
+    // MARK: Lifecycle
+    
+    init(theme: Theme, trackingType: TrackingType) {
+        self.theme = theme
+        self.trackingType = trackingType
+        shareTrackersView = ShareTrackersView(trackingType: trackingType)
+        
+        super.init(nibName: nil, bundle: nil)
+    }
+    
+    @available(*, unavailable)
+    required init(coder: NSCoder) {
+        fatalError()
+    }
+    
+    override func traitCollectionDidChange(_ previousTraitCollection: UITraitCollection?) {
+        super.traitCollectionDidChange(previousTraitCollection)
+        
+        if #available(iOS 13.0, *) {
+            if traitCollection.userInterfaceStyle != previousTraitCollection?.userInterfaceStyle {
+                applyTheme(Theme.of(nil))
+            }
+        }
+    }
+    
+    // MARK: Internal
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        shareTrackersView.actionHandler = { [weak self] action in
+            guard let self = self else { return }
+            
+            switch action {
+            case .didShareWithMailTapped:
+                self.actionHandler?(.takeALookTapped)
```

Comment:
i mentioned in the other PR about considering merging some of those enums as they are very similar to each other

Reply:
Yeah in that PR the new enum used will be unified with TrackingType enum but for action I want to keep it separate because action types are totally different than tracking types even they seem similar. Otherwise I believe it will create confusion.